crates/orderbook/src/app_data.rs

```diff
@@ -60,7 +60,13 @@ impl Registry {
             .insert_full_app_data(&validated.hash, &validated.document)
             .await
         {
-            Ok(()) => Ok((Registered::New, validated.hash)),
+            Ok(()) => {
+                // Post New AppData to IPFS after successful insert.
+                if let Some(ipfs) = &self.ipfs {
+                    let _ = ipfs.post_app_data(validated.clone()).await;
```

Comment:
Could probably make use of the result here, but didn't have any thoughts on how you would like to do it. Maybe we can move the logging out of the inner methods to here...

Reply:
In case this operation is not going to be executed atomically with the DB insert, how about running the IPFS request asynchronously(probably, with a retry ability)?

Reply:
This suggestion sounds a bit like making use of this existing project https://github.com/cowprotocol/ipfs-block-put which can be running as a separate service whose only responsibility receive upload requests and handle them. Then here, we would just send out our requests to upload and forget about it.